annotations/views/rest_views.py

```diff
@@ -321,6 +333,93 @@ def create(self, request, *args, **kwargs):
         headers = self.get_success_headers(serializer.data)
         return Response(reserializer.data, status=status.HTTP_201_CREATED, headers=headers)
 
+    def update(self, request, *args, **kwargs):
+        try:
+            logger.debug(f"Update request from user {request.user.id} for appellation {kwargs.get('pk')}")
+            instance = self.get_object()
+            logger.debug(f"Appellation creator: {instance.createdBy.id}")
+            
+            # Ensure interpretation is provided
+            interpretation_uri = request.data.get('interpretation')
+            if not interpretation_uri:
+                return Response(
+                    {"error": "An appellation must have a concept (interpretation) to be saved."},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+            
+            # Get the position data
+            position_value = request.data.get('position', {}).get('position_value')
+            if position_value:
+                # Update position
+                if not instance.position:
+                    instance.position = DocumentPosition.objects.create(
+                        occursIn_id=request.data['position']['occursIn'],
+                        position_type=request.data['position']['position_type'],
+                        position_value=position_value
+                    )
+                else:
+                    instance.position.position_value = position_value
+                    instance.position.save()
+
+            # Update string representation
+            if 'stringRep' in request.data:
+                instance.stringRep = request.data['stringRep']
+            
+            # Update interpretation
+            try:
+                concept = Concept.objects.get(uri=interpretation_uri)
+                instance.interpretation = concept
+            except Concept.DoesNotExist:
+                return Response(
+                    {"error": "The provided concept (interpretation) does not exist."},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+            
+            instance.save()
+            
+            serializer = self.get_serializer(instance)
+            return Response(serializer.data)
+            
+        except Exception as e:
+            logger.error(f"Error updating appellation: {str(e)}", exc_info=True)
+            return Response(
+                {"error": str(e)},
+                status=status.HTTP_400_BAD_REQUEST
```

Comment:
400 should only be sent when the client could change something to make this request go through, but here you are catching any exception, so it could an exception that has nothing to do with what the client sends? either this should return 500 or different exceptions should be handled differently.